aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,271 @@
+
+        for key in ("calctype", "structure"):
+            if key not in value:
+                return f"required value was not provided for the `{key}` namespace."
+
+        valid_calctypes = {"singlepoint", "geom opt"}
+        if "calctype" in value:
+            if str(value["calctype"].value) not in valid_calctypes:
+                return f"The 'calctype' must be one of {valid_calctypes}, \
+                    but got '{value['calctype']}'."
+
+        if "input_filename" in value:
+            if not str(value["input_filename"].value).endswith(".cif"):
+                return "The parameter 'input_filename' must end with '.cif'"
+
+        # If both structure and calctype are provided, return None
+        return None
+
+    # pylint: disable=too-many-locals
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files for the `Calcjob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+        # Create needed inputs
+        # Define architecture from model if model is given,
+        # otherwise get architecture from inputs and download default model
+        architecture = (
+            str((self.inputs.model).architecture)
+            if self.inputs.model
+            else str(self.inputs.architecture.value)
+        )
+        model_path = (
+            str((self.inputs.model).filepath)
+            if self.inputs.model
+            else ModelData.download(
+                "https://github.com/stfc/janus-core/raw/main/tests/models/mace_mp_small.model",  # pylint:disable=line-too-long
+                architecture,
+            ).filepath
+        )
+
+        # The inputs are saved in the node, but we want their value as a string
+        calctype = str((self.inputs.calctype).value)
+        precision = str((self.inputs.precision).value)
+        device = str((self.inputs.device).value)
+        xyz_filename = str((self.inputs.xyz_output_name).value)
+        input_filename = self.inputs.metadata.options.input_filename
+        log_filename = str((self.inputs.log_filename).value)
+        # Transform the structure data in cif file called input_filename
+        structure = self.inputs.structure
+        cif_structure = structure.get_cif()
+        with folder.open(self._DEFAULT_INPUT_FILE, "w", encoding="utf-8") as inputfile:
+            inputfile.write(cif_structure.get_content())
+
+        cmd_line = {
+            "arch": architecture,
+            "struct": input_filename,
+            "device": device,
+            "log": log_filename,
+            "calc-kwargs": {"model_paths": model_path, "default_dtype": precision},
```

Comment:
```suggestion
            "calc-kwargs": {"model": model_path, "default_dtype": precision},
```
Hopefully works either way